src/pip/_internal/utils/entrypoints.py

```diff
@@ -0,0 +1,30 @@
+import sys
+
+from pip._internal.cli.main import main
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from typing import Optional, List
+
+
+def _wrapper(args=None):
+    # type: (Optional[List[str]]) -> int
+    """Central wrapper for all old entrypoints.
+
+    Historically pip has had several entrypoints defined. Because of issues
+    arising from PATH, sys.path, multiple Pythons, their interactions, and most
+    of them having a pip installed, users suffer every time an entrypoint gets
+    moved.
+
+    To alleviate this pain, and provide a mechanism for warning users and
+    directing them to an appropriate place for help, we now define all of
+    our old entrypoints as wrappers for the current one.
+    """
+    sys.stderr.write(
+        "WARNING: pip is being invoked by an old script wrapper, probably due "
+        "to mixed Python environments or PATH issues. This will fail in a "
+        "future version of pip.\n"
```

Comment:
I'd still like to see the warning suggest that the user try `python -m pip` to avoid the issue, as there's a significant chance that if they are invoking pip via the wrong wrapper, they will be installing files to the wrong Python environment as well....

Reply:
Is "invoke Python with '-m pip'" enough, to avoid the issue mentioned in [#3164 (comment)](https://github.com/pypa/pip/issues/3164#issuecomment-406899921)?

Reply:
Sounds fair to me. Personally, I'm inclined to hope that anyone using pip is clued up enough to infer from a suggestion to use `python` that they should use whatever command they would normally use to run Python, but let's avoid the risk. We can always change the wording if people get confused by "invoke Python with `-m pip`"...

Reply:
Updated. Now the error message looks like

```
WARNING: pip is being invoked by an old script wrapper. This will fail in a future version of pip.
Please see https://github.com/pypa/pip/issues/5599 for advice on fixing the underlying issue.
To avoid this problem you can invoke Python with '-m pip' instead of running pip directly.
```